deathcause.lua

```diff
@@ -0,0 +1,212 @@
+    'experienced death by BURN_ALIVE',    -- BURN_ALIVE
+    'experienced death by FEED_TO_BEASTS',    -- FEED_TO_BEASTS
+    'experienced death by HACK_TO_PIECES',    -- HACK_TO_PIECES
+    'choked on air',                -- LEAVE_OUT_IN_AIR
+    'experienced death by BOIL',    -- BOIL
+    'melted',                        -- MELT
+    'experienced death by CONDENSE',    -- CONDENSE
+    'experienced death by SOLIDIFY',    -- SOLIDIFY
+    'succumbed to infection',        -- INFECTION
+    "s ghost was put to rest with a memorial",    -- MEMORIALIZE
+    'scared to death',                -- SCARE
+    'experienced death by DARKNESS',    -- DARKNESS
+    'experienced death by COLLAPSE',    -- COLLAPSE
+    'was drained of blood',            -- DRAIN_BLOOD
+    'was slaughtered',                -- SLAUGHTER
+    'became roadkill',                -- VEHICLE
+    'killed by a falling object',    -- FALLING_OBJECT
+}
+
+-- Creates a table of all items at the given location optionally matching a given item type
+function item_find(pos_x, pos_y, pos_z, item_type)
+    local items = {}
+    for _, item in ipairs(df.global.world.items.all) do
+        if item.pos.x == pos_x and item.pos.y == pos_y and item.pos.z == pos_z then
+            if not item_type or item:getType() == item_type then
+                table.insert(items, item)
+            end
+        end
+    end
+    return items
+end
+
+-- Finds a unit with the given id or nil if no unit is found
+function unit_find(unit_id)
+    return utils.binsearch(df.global.world.units.all, unit_id, 'id')
+end
+
+function getRace(race_id)
+    return df.global.world.raws.creatures.all[race_id]
+end
+
+function getRaceNameSingular(race_id)
+    return getRace(race_id).name[0]
+end
+
+function getDeathStringFromCause(cause)
+    if cause == -1 then
+        return "none"
+    else
+        return DEATH_TYPES[cause]
+    end
+end
+
+function display_death_unit(unit)
+    local str = string.format("The %s", getRaceNameSingular(unit.race))
+    if unit.name.has_name then
+        str = str .. string.format(" %s", dfhack.TranslateName(unit.name))
+    end
+
+    if not unit.flags2.killed and not unit.flags3.ghostly then
+        str = str .. " is not dead yet!"
+        print(str)
+        return
+    end
+
+    str = str .. string.format("%s", getDeathStringFromCause(unit.counters.death_cause))
+
+    local incident = df.incident.find(unit.counters.death_id)
+    if incident then
+        str = str .. string.format(" in year %d", incident.event_year)
+    end
+
+    if incident and incident.criminal then
+        local killer = df.unit.find(incident.criminal)
+        if killer then
+            str = str .. string.format(" killed by the %s", getRaceNameSingular(killer.race))
+            if killer.name.has_name then
+                str = str .. string.format(" %s", dfhack.TranslateName(killer.name))
+            end
+        end
+    end
+
+    str = str:gsub(",%s*$", "")
```

Comment:
when does this come up? I don't see you inserting any commas.

Reply:
It might not come up anymore, will take a look